services/QuillLMS/spec/queries/progress_reports/district_standards_reports_spec.rb

```diff
@@ -3,308 +3,194 @@
 
-        context 'student2 has finished activity_sessions with activity1' do
-          before do
-            create(
-              :activity_session,
-              :finished,
-              activity: activity1,
-              percentage: percentage2,
-              timespent: timespent2,
-              classroom_unit: classroom_unit,
-              user: student2
-            )
-          end
+      expect(results.is_a?(Array)).to be true
+      expect(results.count).to eq 1
+      expect(results.first.keys).to match_array(payload_keys)
+    end
 
-          it { expects_results }
+    it 'should handle multiple standards' do
+      create(:standard, id: 6)
+      stubbed_payload_sequence = [example_json, example_json2]
 
-          it do
-            expect(subject.size).to eq 1
-            expect(subject[0]["total_activity_count"]).to eq 1
-            expect(subject[0]["total_student_count"]).to eq 2
-          end
-        end
+      allow(QuillBigQuery::Runner).to receive(:get_response).and_return(*stubbed_payload_sequence)
 
-        context 'student2 has finished activity_sessions with activity2' do
-          before do
-            create(
-              :activity_session,
-              :finished,
-              activity: activity2,
-              percentage: percentage2,
-              timespent: timespent2,
-              classroom_unit: classroom_unit,
-              user: student2
-            )
-          end
+      report = described_class.new(admin.id)
+      allow(report).to receive(:user_ids_query).and_return '3945512'
 
-          it { expects_results }
+      results = report.results
 
-          it do
-            expect(subject.size).to eq 1
-            expect(subject[0]["total_activity_count"]).to eq 2
-            expect(subject[0]["total_student_count"]).to eq 2
-          end
-        end
+      expect(results.pluck('id')).to match_array [6,7]
+      expect(results.count).to eq 2
+      results.each do |result|
+        expect(result.keys).to match_array(payload_keys)
       end
     end
-  end
-
-  def finished_activity_sessions
-    ActivitySession.where(is_final_score: true).to_a
-  end
 
-  def finished_standard_activity_sessions(standard)
-    finished_activity_sessions.select { |as| as.activity.standard == standard }
-  end
-
-  def finished_standard_activities(standard)
-    finished_standard_activity_sessions(standard).map(&:activity).uniq
-  end
-
-  def finished_standard_student_ids(standard)
-    finished_standard_activity_sessions(standard).pluck(:user_id).uniq
-  end
-
-  def proficient_standard_student_ids(standard)
-    finished_standard_activity_sessions(standard).select { |as| as.percentage >= 0.80 }.pluck(:user_id).uniq
-  end
+    it 'should handle bigquery responses with null rows' do
+      create(:standard, id: 6)
+      stubbed_payload_sequence = [example_json, exmaple_null_json]
+      allow(QuillBigQuery::Runner).to receive(:get_response).and_return(*stubbed_payload_sequence)
```

Comment:
This is rarely used in the codebase, but passing an array to an `allow` chain allows sequencing of stubbed return values.